virttest/utils_windows/drive.py

```diff
@@ -37,3 +39,93 @@ def get_floppy_drives_letter(session):
     """
     cond = "MediaType!=0 AND MediaType!=11 AND MediaType!=12"
     return _logical_disks(session, cond=cond, props=["DeviceID"])
+
+
+def rescan_disk(session):
+    """
+    Rescan disk in windows guest.
+
+    :param session: Session object.
+    """
+    rescan_cmd = "echo rescan > cmd"
+    rescan_cmd += " && echo exit >> cmd"
+    rescan_cmd += " && diskpart /s cmd"
+    session.cmd(rescan_cmd)
+
+
+def get_drive_volume_id(session, did, disk_letter):
+    """
+    Get a drive volume id in windows guest.
+
+    :param session: Session object.
+    :param did: Disk index.
+    :param disk_letter: Destination volume location.
+
+    :return: volume id, cmd header and cmd footer
+    """
+    list_disk_cmd = "echo list disk > cmd && "
+    list_disk_cmd += "echo exit >> disk && diskpart /s cmd"
+    disks = session.cmd_output(list_disk_cmd, timeout=120)
+
+    for disk in disks.splitlines():
+        if re.search(r"DISK %s" % did, disk, re.I | re.M):
+            cmd_header = 'echo list disk > cmd &&'
```

Comment:
Is there any reason to list all the disks again?

Reply:
Canceled list again on this func, thanks~